.github/workflows/ignition-workflow.yml

```diff
@@ -0,0 +1,225 @@
+name: Build and pack Ignition packages
+on:
+  workflow_call:
+    inputs:
+      deploy:
+        required: false
+        type: string
+        default: 'false'
+      release:
+        required: false
+        type: string
+        default: 'false'
+
+    secrets:
+      auto_commit_user:
+        required: true
+      auto_commit_mail:
+        required: true
+      auto_commit_pwd:
+        required: true
+      nexus_publisher_user:
+        required: true
+      nexus_publisher_password:
+        required: true
+      gh_token:
+        required: true
+
+jobs:
+  Build:
+    if: ${{ inputs.release == 'false' }}
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Find Package details
+      id: vars
+      run: |
+        CMAKE_LISTS="CMakeLists.txt"
+        PACKAGE_NAME=$(grep -m 1 "project(" "$CMAKE_LISTS" | sed -n 's/project(\(.*\) VERSION \(.*\))/\1/p')
+        PACKAGE_VERSION=$(grep -m 1 "project(" "$CMAKE_LISTS" | sed -n 's/project(\(.*\) VERSION \(.*\))/\2/p')
+        echo "pkg_name=$PACKAGE_NAME" >> $GITHUB_OUTPUT
+        echo "pkg_version=$PACKAGE_VERSION" >> $GITHUB_OUTPUT
+
+    - name: Raise Version
+      id: vars-after-raise
+      run: |
+        PACKAGE_VERSION=${{ steps.vars.outputs.pkg_version }}
+        IFS='.' read major_version minor_version patch_version <<< "$PACKAGE_VERSION"
+        main_version="$major_version.$minor_version"
+        raised_patchversion=$(($patch_version+1))
+        RAISED_PACKAGE_VERSION="$main_version.$raised_patchversion"
+        echo "pkg_version=$RAISED_PACKAGE_VERSION" >> $GITHUB_OUTPUT
+    
+    - name: Update Version
+      run: |
+        NEW_VERSION=${{ steps.vars-after-raise.outputs.pkg_version }}
+        PACKAGE_NAME=${{ steps.vars.outputs.pkg_name }}
+        sed -i "s/\(project($PACKAGE_NAME VERSION \)[0-9.]*\()\)/\1$NEW_VERSION\2/" CMakeLists.txt
+
+    - name: Compile, test and package
+      id: ci
+      uses: ignition-tooling/action-ignition-ci@focal
+      with:
+        cpplint-enabled: true
+
+    - name: Archive debian artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: packages
+        path: artifacts/*
+        retention-days: 5
+
+    - name: Print generated packages
+      run: ls -la artifacts
+
+  Publish:
+    if: ${{ inputs.deploy == 'true' }}
+    runs-on: ubuntu-latest
+    needs:  [Build]
```

Comment:
what if `inputs.release != 'false'` and `inputs.deploy == 'true'` , Build will not be executed but it is a dependency , no ?